autogpt_platform/backend/backend/executor/manager.py

```diff
@@ -457,25 +433,14 @@ def on_node_executor_stop(cls):
 
         logger.info(f"[on_node_executor_stop {cls.pid}] ⏳ Releasing locks...")
         cls.creds_manager.release_all_locks()
-        logger.info(f"[on_node_executor_stop {cls.pid}] ⏳ Disconnecting DB...")
-        cls.loop.run_until_complete(db.disconnect())
         logger.info(f"[on_node_executor_stop {cls.pid}] ⏳ Disconnecting Redis...")
         redis.disconnect()
         logger.info(f"[on_node_executor_stop {cls.pid}] ✅ Finished cleanup")
 
     @classmethod
     def on_node_executor_sigterm(cls):
         llprint(f"[on_node_executor_sigterm {cls.pid}] ⚠️ SIGTERM received")
-        if not cls.shutdown_lock.acquire(blocking=False):
-            return  # already shutting down, no need to self-terminate
-
-        llprint(f"[on_node_executor_sigterm {cls.pid}] ⏳ Releasing locks...")
-        cls.creds_manager.release_all_locks()
-        llprint(f"[on_node_executor_sigterm {cls.pid}] ⏳ Disconnecting DB...")
-        cls.loop.run_until_complete(db.disconnect())
-        llprint(f"[on_node_executor_sigterm {cls.pid}] ⏳ Disconnecting Redis...")
-        redis.disconnect()
-        llprint(f"[on_node_executor_sigterm {cls.pid}] ✅ Finished cleanup")
+        cls.on_node_executor_stop()
```

Comment:
Please revert this because it will break

Reply:
Any explanation so I can add it as comment for the code duplication ?

Reply:
Also why it doesn't break the integration test /

Reply:
ohh is it the llprint that's breaking ?

Reply:
`logger.info` isn't allowed in signal handlers, hence the `llprint`. Trying to use the logger causes intermittent low-level errors.